server/portal/views/base.py

```diff
@@ -27,52 +26,62 @@ def dispatch(self, request, *args, **kwargs):
+        except (PermissionDenied, Http404) as e:
+            # log information but re-raise exception to let django handle response
+            logger.error(e, exc_info=True)
+            raise e
+        except ApiException as e:
+            status = e.response.status_code
+            message = e.response.reason
+            extra = e.extra
             if status != 404:
                 logger.error(
                     '%s: %s',
                     message,
-                    err.response.text,
+                    e.response.text,
                     exc_info=True,
                     extra=extra
                 )
             else:
                 logger.info('Error %s', message, exc_info=True, extra=extra)
             return JsonResponse({'message': message}, status=400)
-        except (ConnectionError, HTTPError) as err:
-            status = err.response.status_code
-            content = err.response.json()
-            message = content.get("message", "Unknown Error")
-            if status in [404, 403]:
-                logger.warning(
-                    '%s: %s',
-                    message,
-                    err.response.text,
-                    exc_info=True,
-                    extra={
-                        'username': request.user.username,
-                        'session_key': request.session.session_key
-                    }
-                )
+        except (ConnectionError, HTTPError) as e:
+            if e.response is not None:
+                status = e.response.status_code
+                try:
+                    content = e.response.json()
+                    message = content.get("message", "Unknown Error")
+                except ValueError:
+                    message = "Unknown Error"
+                if status in [404, 403]:
+                    logger.warning(
+                        '%s: %s',
+                        message,
+                        e.response.text,
+                        exc_info=True,
+                        extra={
+                            'username': request.user.username,
+                            'session_key': request.session.session_key
+                        }
+                    )
+                else:
+                    logger.error(
+                        '%s: %s',
+                        message,
+                        e.response.text,
+                        exc_info=True,
+                        extra={
+                            'username': request.user.username,
+                            'session_key': request.session.session_key
+                        }
+                    )
             else:
-                logger.error(
-                    '%s: %s',
-                    message,
-                    err.response.text,
-                    exc_info=True,
-                    extra={
-                        'username': request.user.username,
-                        'session_key': request.session.session_key
```

Comment:
Will this extra info be missed?

Reply:
it is still provided in line 62 and 73 of change file.  in line 79 of changed file, it is not there.  do you want me to add it? 
there would need to be a check that those extra stuff exists (like `user`) before adding it.

Reply:
But 62/73 are a part of a different if/else block, no? I think it may still be worth including

Reply:
yes,  that's a good point.  will add.

Reply:
@rstijerina , i added similar extra information to the spot discussed above.  **Note** though that this extra information (`username` + `session_key`) are not logged in our log formatters (i.e. in settings.py).  I propose a follow up JIRA ticket for the backend epic to clean that up.  Cool, that i create that?  It would be contain notes about these logging statements plus our formatters.

Reply:
Nice find, yeah sounds great 👍

Reply:
Added ticket https://jira.tacc.utexas.edu/browse/FP-796